File: javer/src/main/java/com/uuhnaut69/javer/resource/RestaurantResource.java
package com.uuhnaut69.javer.resource;

import com.uuhnaut69.javer.model.Dish;
import com.uuhnaut69.javer.model.Restaurant;
import com.uuhnaut69.javer.resource.dto.DishRequest;
import com.uuhnaut69.javer.resource.dto.RestaurantRequest;
import com.uuhnaut69.javer.service.DishService;
import com.uuhnaut69.javer.service.RestaurantService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.*;

import javax.validation.Valid;
import java.util.List;
import java.util.UUID;

@RestController
@RequiredArgsConstructor
@RequestMapping("/api/restaurants")
public class RestaurantResource {

  private final DishService dishService;

  private final RestaurantService restaurantService;

  @PostMapping
  @ResponseStatus(HttpStatus.CREATED)
  public Restaurant create(@RequestBody @Valid RestaurantRequest restaurantRequest) {
    return restaurantService.create(restaurantRequest);
  }

  @GetMapping
  public List<Restaurant> findAll() {
    return restaurantService.findAll();
  }

  @PostMapping("/{restaurantId}")
  @ResponseStatus(HttpStatus.CREATED)
  public Dish create(@PathVariable UUID restaurantId, @RequestBody @Valid DishRequest dishRequest) {
    return dishService.create(restaurantId, dishRequest);
  }

  @GetMapping("/{restaurantId}")
  public List<Dish> findAllByRestaurantId(@PathVariable UUID restaurantId) {
    return dishService.findAllByRestaurantId(restaurantId);
  }
}

File: api/kover/src/main/kotlin/com/uuhnaut69/kover/resource/RestaurantResource.kt
package com.uuhnaut69.kover.resource

import com.uuhnaut69.kover.model.Dish
import com.uuhnaut69.kover.model.Restaurant
import com.uuhnaut69.kover.resource.dto.DishRequest
import com.uuhnaut69.kover.resource.dto.RestaurantRequest
import com.uuhnaut69.kover.service.DishService
import com.uuhnaut69.kover.service.RestaurantService
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.*
import java.util.*
import javax.validation.Valid

@RestController
@RequestMapping("/api/restaurants")
class RestaurantResource(private val dishService: DishService, private val restaurantService: RestaurantService) {

    @PostMapping
    @ResponseStatus(HttpStatus.CREATED)
    fun createRestaurant(@RequestBody @Valid restaurantRequest: RestaurantRequest): Restaurant =
        restaurantService.create(restaurantRequest)

    @GetMapping
    fun findAllRestaurants(): List<Restaurant> = restaurantService.findAll()

    @PostMapping("/{restaurantId}")
    @ResponseStatus(HttpStatus.CREATED)
    fun createDish(@PathVariable restaurantId: UUID, @RequestBody @Valid dishRequest: DishRequest): Dish =
        dishService.create(restaurantId, dishRequest)

    @GetMapping("/{restaurantId}")
    fun findAllByRestaurantId(@PathVariable restaurantId: UUID): List<Dish> =
        dishService.findAllByRestaurantId(restaurantId)
}